docs/DeveloperGuide.md

```diff
@@ -2,36 +2,80 @@
 
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+This developer guide follows the suggested format given in [AddressBook-Level3 developer guide](https://se-education.org/addressbook-level3/DeveloperGuide.html).
 
-## Design & implementation
+## Design
+### Architecture
+![Architecture diagram](Architecture_diagram.jpg)
```

Comment:
Very well done Architecture Diagram. Adding a class diagram below would be useful for a more in-depth illustration of relationships between classes.